interpreter/value.go

```diff
@@ -254,6 +255,7 @@ type ValueIteratorContext interface {
 // ValueIterator is an iterator which returns values.
 // When Next returns nil, it signals the end of the iterator.
 type ValueIterator interface {
+	HasNext() bool
```

Comment:
Needed this to make the looping instructions working (copied over from `vm.ValueIterator`)